src/Update/Nix/FetchGit.hs

```diff
@@ -128,48 +72,44 @@ extractFetchGitArgs = \case
   e -> Left (ArgNotASet e)
 
 --------------------------------------------------------------------------------
--- Massaging the data a little to make it easier to process
+-- Getting updated information from the internet.
 --------------------------------------------------------------------------------
 
--- | The info needed to find the latest git version and the info about where to
--- update the rev and sha256.
-data FetchGitUpdateInfo = FetchGitUpdateInfo{ urlString :: Text
-                                            , revPos    :: SourceSpan
-                                            , sha256Pos :: SourceSpan
-                                            }
-  deriving (Show)
-
--- | Given some arguments to fetchgit, extract the information necessary to
--- update it
-fetchGitArgsToUpdate :: Text -> FetchGitArgs -> Either Warning FetchGitUpdateInfo
-fetchGitArgsToUpdate t as = FetchGitUpdateInfo (extractUrlString (repoLocation as))
-                                           <$> exprSpan t (revExpr as)
-                                           <*> exprSpan t (sha256Expr as)
+getFetchGitLatestInfo :: FetchGitArgs -> IO (Either Warning FetchGitLatestInfo)
+getFetchGitLatestInfo args = do
+  result <- nixPrefetchGit (extractUrlString $ repoLocation args)
+  case result of
+    Left e -> pure $ Left e
+    Right o -> return $ FetchGitLatestInfo args (rev o) (sha256 o)
```

Comment:
Is it possible to use `pure` instead of `return` everywhere to keep things consistent.